File: tobas_core/tobas_common/tobas_ros2_tools/src/time.cpp
// SPDX-License-Identifier: GPL-3.0-or-later
// Copyright (C) 2026 Tobas, Inc.

#include "tobas_ros2_tools/time.hpp"

namespace ch = std::chrono;

namespace tobas
{
namespace ros2
{
namespace
{
constexpr auto kNanosecondsPerSecond = 1'000'000'000;
}  // namespace

void timeChronoToMsg(const ch::steady_clock::duration& c, builtin_interfaces::msg::Time& m)
{
  const auto nsec = c.count();
  assert(nsec >= 0);
  m.sec = nsec / kNanosecondsPerSecond;
  m.nanosec = nsec % kNanosecondsPerSecond;
}
}  // namespace ros2
}  // namespace tobas

File: tobas_core/tobas_common/tobas_std_tools_py/tobas_std_tools_py/time.py
# SPDX-License-Identifier: GPL-3.0-or-later
# Copyright (C) 2026 Tobas, Inc.

from typing import Tuple


def get_h_m_s(sec: int) -> Tuple[int, int, int]:
    """Convert `sec` to hour, minute, and second."""

    m, s = divmod(sec, 60)
    h, m = divmod(m, 60)
    return h, m, s
